test/tpmclient/tpmclient.int.c

```diff
@@ -100,40 +79,36 @@ TSS2_TCTI_CONTEXT *resMgrTctiContext = 0;
 #define YES 1
 #define NO 0
 
-#define MSFT_MANUFACTURER_ID 0x4d534654
-#define IBM_MANUFACTURER_ID 0x49424d20
-#define LEVEL_STRING_SIZE 50
-
-static void ErrorHandler( UINT32 rval )
+static void ErrorHandler(UINT32 rval, char *errorString, int errorStringSize)
 {
     UINT32 errorLevel = rval & TSS2_RC_LAYER_MASK;
-    char levelString[LEVEL_STRING_SIZE + 1];
+    char levelString[32];
```

Comment:
You changed levelString from LEVEL_STRING_SIZE+1 to 32, then use strncpy() of LEVEL_STRING_SIZE (defined as 50).
So this allows for buffer overflows.